inbm/diagnostic-agent/diagnostic/event_watcher.py

```diff
@@ -51,6 +50,7 @@ def run_docker_bench_security(self):  # pragma: no cover
         """Launch Docker Bench Security in separate thread."""
         def run():
             if current_dbs_mode != ConfigDbs.OFF:
+                self.lock.acquire()
```

Comment:
What happens when the thread is waiting here? Does it block a caller or message sender upstream?

Reply:
It will not block anything it will be in queue. Before executing run method, it can acquire the lock. 
After executing run method, it releases the lock. 
Other thread will wait for the lock to be released before they executing the run method.